Highway.cpp

```diff
@@ -22,13 +37,28 @@ void Highway::addVehicleInternal(Vehicle* v)
 
 void Highway::removeVehicleInternal(Vehicle* v)
 {
-    assert(false);
+    // assert(false);
 
     /*
     depending on the derived type, call the member function that tries to evade the cops. 
 
     trucks pull over, but cars and bikes try to evade!!
     */
+
+    if (Car* car = dynamic_cast<Car*>(v))
+    {
+        car->tryToEvade();
+    }
+
+    if (Motorcycle* motorcycle = dynamic_cast<Motorcycle*>(v))
+    {
+        motorcycle->tryToEvade();
+    }
+
+    if (SemiTruck* semiTruck = dynamic_cast<SemiTruck*>(v))
+    {
+        semiTruck->pullOver();
+    }
```

Comment:
use `else-if`
you're still checking if it's a Motorcycle, even if it's a Car
- [ ] revise